packages/DApp/src/hooks/useCommunities.ts

```diff
@@ -23,57 +27,96 @@ export function useCommunities(publicKeys: string[]) {
+  //         }
+  //       } else {
+  //         return undefined
+  //       }
+  //     })
+  //   )
+  // }, [communitiesDetails, featureVotes, JSON.stringify(publicKeys)])
 
   useEffect(() => {
-    publicKeys.forEach((publicKey, idx) => {
-      if (publicKey && communitiesHistories[idx]) {
-        const setCommunity = async () => {
-          const communityDetail = await getCommunityDetails(publicKey)
-          if (communityDetail) {
-            const communityHistory = communitiesHistories[idx]?.[0]
-            if (communityHistory && communityHistory.length > 0) {
-              const votingHistory = communityHistory.map((room: any) => {
-                const endAt = new Date(room.endAt.toNumber() * 1000)
-                return {
-                  ID: room.roomNumber.toNumber(),
-                  type: room.voteType === 1 ? 'Add' : 'Remove',
-                  result:
-                    endAt > new Date()
-                      ? 'Ongoing'
-                      : room.totalVotesFor.gt(room.totalVotesAgainst)
-                      ? 'Passed'
-                      : 'Failed',
-                  date: endAt,
-                }
-              })
-              dispatch({ ...communityDetail, votingHistory })
-            } else {
-              dispatch({ ...communityDetail, votingHistory: [] })
-            }
+    if (!waku || publicKeys.length === 0) return
+
+    const fetch = async () => {
+      await Promise.all(
+        publicKeys.map(async (publicKey) => {
+          const deserializedPublicKey = deserializePublicKey(publicKey)
+
+          if (communitiesDetails[deserializedPublicKey]) {
+            return
           }
-        }
-        setCommunity()
-      }
-    })
-  }, [JSON.stringify(publicKeys), communitiesHistories])
 
-  useEffect(() => {
-    setReturnCommunities(
-      publicKeys.map((publicKey) => {
-        const detail = communitiesDetails[publicKey]
-        if (detail) {
-          if (featureVotes[publicKey]) {
-            return { ...detail, featureVotes: featureVotes[publicKey].sum }
-          } else {
-            return { ...detail, featureVotes: BigNumber.from(0) }
+          const requestClient = new RequestClient(waku)
+          const community = await requestClient.fetchCommunityDescription(deserializedPublicKey)
+
+          if (!community) {
+            console.warn(`Community ${deserializedPublicKey} not found`)
+            return
           }
-        } else {
-          return undefined
-        }
-      })
-    )
-  }, [communitiesDetails, featureVotes, JSON.stringify(publicKeys)])
 
-  return returnCommunities
+          dispatch({
+            publicKey: deserializedPublicKey,
+            name: community!.identity!.displayName,
+            description: community!.identity!.description,
+            ens: community!.identity!.ensName,
+            icon: URL.createObjectURL(
+              new Blob([community!.identity!.images.large.payload], {
```

Comment:
`images` can be undefined, unset by user

<img width="225" alt="image" src="https://user-images.githubusercontent.com/13265126/236072323-24d3544c-e07b-4bc2-a2cf-c3196dc2ec4c.png">